src/components/CustomerQuota/CustomerQuotaProxy.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FunctionComponent, useContext, useCallback } from "react";
+import { NavigationProps, Policy } from "../../types";
+import { ProductContext, ProductContextValue } from "../../context/products";
+import { CustomerQuotaScreen } from "./CustomerQuotaScreen";
+
+export const CustomerQuotaProxy: FunctionComponent<NavigationProps> = ({
+  navigation
+}) => {
+  // coming from NRIC screen, it will be a string
+  // coming from appeal, it can be an array if group appeal is supported
+  const navId = navigation.getParam("id");
+  const navIds = Array.isArray(navId) ? navId : [navId];
+
+  const selectedProducts: Policy[] = navigation.getParam("products");
+
+  // we have to replace getProduct as well. Because all the child component is invoking getProduct and it will retrieve from the parent context.
+  // the products at the parent context is not what we are interested in
+  const getProduct = useCallback(
+    (category: string): Policy | undefined => {
+      return selectedProducts.find(product => product.category === category);
+    },
+    [selectedProducts]
+  );
```

Comment:
If we do not redefine `getProduct`, that `getProduct` function will be getting its `products` from the parent context. Although we have replace the `products` in the `CustomerQuotaProxy`, this replacement is not visible in the parent context when `getProduct` is being invoked

Most of the child component is relying on this function `getProduct` to retrieve it.. if we are working with the `products` array directly, then it can work in the child component